truffle/src/main/ruby/core/rubinius/api/shims/tuple.rb

```diff
@@ -28,6 +22,48 @@ def copy_from(other, start, length, dest)
 
 class Array
 
+  def copy_from(other, start, length, dest)
+    # Rubinius.primitive :tuple_copy_from
```

Comment:
We have the primitive node classes in `org.jruby.truffle.nodes.rubinius`, but surprisingly, we don't seem to have an `Array` one yet.  That's where I'd advocate putting it.

The primitive name itself forms the actual namespace, so it doesn't need to be tied to a particular class . . . that's just for organization. As an example, `Rubinius::Character` is a subclass of `String`, much like the `Tuple` and `Array` relationship.  But I recently added a "character" primitive to the `StringPrimitiveNodes` class.